### PR TITLE
feat: add `<PrismicLink>` component

### PR DESCRIPTION
**Pull request by samlfair**
## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Add a `<PrismicLink>` component that can accept a Link field, Content Relationship field, or entire Document, and output an `<a>` element.

`<PrismicLink>` accepts all [SvelteKit link options](https://kit.svelte.dev/docs/link-options) and all [HTML link attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) as props. It also accepts a Link Resolver.

## Checklist:

- [x] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

puppies

### CODE REVIEW

**Review by angeloashmore — CHANGES_REQUESTED** [review]
Nice set of improvements to `<PrismicLink>`! 😃

We can make some optimization to the code to simplify its logic and readability. I left detailed comments on the code, but in summary:

- Use `$$Props` to extend Svelte's native anchor props.
- Compute `rel` and `target` using simpler logic.
- Look for ways to write less code with the same output, sometimes taking advantage of default behaviors.

Let me know if you have any questions or would like to work on a refactor together. 🙂

File: src/lib/components/PrismicLink.svelte [hunk 1]
```diff
@@ -1,67 +1,182 @@
 <script lang="ts">
 	import * as prismicH from "@prismicio/helpers";
 	import type * as prismicT from "@prismicio/types";
+	// import { usePrismic } from "../usePrismic";
```
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:4]: ✅

File: src/lib/components/PrismicLink.svelte [hunk 2]
```diff
@@ -1,67 +1,182 @@
...
+	function setTarget(field: LinkFieldOrDocument, target: Target) {
+		if(target) {
+			return target
+		}
+		if("target" in field) {
+			return field.target
+		}
+
+		return ""
+	}
```
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:143]: ✅

File: src/lib/components/PrismicLink.svelte [hunk 3]
```diff
@@ -1,67 +1,182 @@
...
+		target: setTarget(field, target),
+		rel: setRel(field, rel, target),
 		type,
 	};
 
 	const link: string | null = field
-		? prismicH.asLink(field, linkResolver || usePrismic().linkResolver)
+		? prismicH.asLink(field, linkResolver)
 		: "";
 	const href: string = link || "";
```
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:158]: ✅

File: src/lib/components/PrismicLink.svelte [hunk 4]
```diff
@@ -1,67 +1,182 @@
...
+
 <a
 	{href}
 	{...attrs}
-	sveltekit:prefetch={prefetch}
-	sveltekit:noscroll={noscroll}
+	data-sveltekit-preload-data={sveltekitPreloadData || "hover"}
+	data-sveltekit-preload-code={sveltekitPreloadCode || "hover"}
+	data-sveltekit-reload={sveltekitReload || "off"}
+	data-sveltekit-noscroll={sveltekitNoscroll || "off"}
```
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:179]: Yep, this looks good to me.

File: src/lib/components/PrismicLink.svelte [hunk 5]
```diff
@@ -1,67 +1,182 @@
...
+	```svelte
+		<PrismicLink field={document.data.example_link}>
+			Example anchor text.
+		</PrismicLink>
+  ```
+-->
+
 <a
 	{href}
 	{...attrs}
```
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:175]: Nope, I think you're right that `...$$restProps` handles this much more clenly.

File: src/lib/components/PrismicLink.svelte [hunk 6]
```diff
@@ -1,67 +1,182 @@
...
+
+	/**
+	 * The Link Resolver used to resolve links.
+	 *
+	 * @remarks
+	 * If your app uses Route Resolvers when querying for your Prismic
+	 * repository's content, a Link Resolver does not need to be provided.
+	 * @see Learn about Link Resolvers and Route Resolvers {@link https://prismic.io/docs/core-concepts/link-resolver-route-resolver}
+	 */
+	export let linkResolver: prismicH.LinkResolverFunction | null | undefined = null
```
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:23]: Good to know! I learned the opposite in bootcamp, but I'm happy to follow this rule ✅

File: src/lib/components/PrismicLink.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (samlfair) [reply src/lib/components/PrismicLink.svelte:175]: ✅

**Comment by samlfair** [comment]
@angeloashmore Thanks again for your feedback. I think I've addressed everything, and I responded to your question about `rel="external"`. Let me know what you think. When things are looking good, I can merge and publish this.

File: src/lib/components/PrismicLink.svelte [hunk 7]
```diff
@@ -1,67 +1,182 @@
...
+				const defaultBlankTargetRelAttributes = ["noopener", "noreferrer"];
+				defaultBlankTargetRelAttributes.forEach((item) => {
+					if (!rels.includes(item)) {
+						rels.push(item)
+					};
+				});
+			}
+			if (!rels.includes("external")) {
+				rels.push("external");
+			}
```
[the remark this answers is not in the document]
Comment (angeloashmore) [reply src/lib/components/PrismicLink.svelte]: 👍 Sounds good to me. It was a question of, "Will SvelteKit work normally without `rel="external"`?"

With the updated code, developers can add `rel="external"` manually, so I think we have all of our bases covered.

**Review by angeloashmore — APPROVED** [review]

**Comment by angeloashmore** [comment]
It looks great to me! Let's merge and publish. 🙂

**Comment by samlfair** [comment]
Published 🚀

**Pull request merged**